commons-core/src/main/scala/com/avsystem/commons/rpc/AsRealRaw.scala

```diff
@@ -0,0 +1,54 @@
+package com.avsystem.commons
+package rpc
+
+import scala.annotation.implicitNotFound
+
+@implicitNotFound("don't know how to encode ${T} as ${R}, appropriate AsRaw instance not found")
+trait AsRaw[R, T] {
```

Comment:
Could you replace `R` and `T` with more meaningful names? Maybe `Raw` and `Real`?